### PR TITLE
[release/2.1.3xx] Replace alpine with linux-musl builds

### PR DESCRIPTION
**Pull request by weshaggard**
Port of change https://github.com/dotnet/cli/pull/9104

cc @eerhardt @livarcocc @johnbeisner

### CODE REVIEW

File: scripts/obtain/dotnet-install.sh [hunk 1]
```diff
@@ -166,9 +162,12 @@ get_current_os_name() {
         local linux_platform_name
         linux_platform_name="$(get_linux_platform_name)" || { echo "linux" && return 0 ; }
 
-        if [[ $linux_platform_name == "rhel.6" || $linux_platform_name == "alpine.3.6" ]]; then
+        if [[ $linux_platform_name == "rhel.6" ]]; then
             echo $linux_platform_name
             return 0
+        elif [[ $linux_platform_name == alpine* ]]; then
+            echo "linux-musl"
```
Comment (eerhardt) [inline scripts/obtain/dotnet-install.sh]: This isn't really going to work until we get the first build out right?  Doesn't "stage0" use this script?  So if we change this before we actually have a build that is `linux-musl`, it is going to fail the first time we try building on alpine.

Comment (eerhardt) [reply scripts/obtain/dotnet-install.sh]: Looks like "yes":

```
13:26:39 + /opt/code/scripts/obtain/dotnet-install.sh --version 2.2.0-preview1-007799 --install-dir /opt/code/.dotnet_stage0/x64 --architecture x64
13:26:39 dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.0-preview1-007799/dotnet-sdk-2.2.0-preview1-007799-linux-musl-x64.tar.gz
13:26:39 Connecting to dotnetcli.azureedge.net (72.21.81.200:443)
13:26:39 wget: server returned error: HTTP/1.1 404 Not Found
13:26:39 dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.0-preview1-007799/dotnet-sdk-2.2.0-preview1-007799-linux-musl-x64.tar.gz
13:26:39 dotnet_install: Error: Could not download .NET Core SDK version 2.2.0-preview1-007799
13:26:39 
```

Comment (weshaggard) [reply scripts/obtain/dotnet-install.sh]: LOL... yep I just hit that (https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1614067&_a=summary) in my testing so I think we will have to keep it as alpine for now until we get a build of it.

Comment (johnbeisner) [reply scripts/obtain/dotnet-install.sh]: @eerhardt 
Whichever drop we are consuming [https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.0-preview1-007799] will need a SDK with the new naming convention; each time we introduce a new RID - a 'bootstrap' needs to occur.

Comment (weshaggard) [reply scripts/obtain/dotnet-install.sh]: If it turns out we need to manually put one up I will do that, for my current test run I just switch to using alpine.

Comment (johnbeisner) [reply scripts/obtain/dotnet-install.sh]: 1. https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.0-preview1-007799/dotnet-sdk-2.2.0-preview1-007799-linux-musl-x64.tar.gz
2. NuPkg: runtime.linux-musl-x64.Microsoft.NETCore.DotNetHostResolver (>= 2.1.0-rc1-26419-02)

Comment (johnbeisner) [reply scripts/obtain/dotnet-install.sh]: Uploaded: https://dotnetcli.azureedge.net/dotnet/Sdk/2.2.0-preview1-007799/dotnet-sdk-2.2.0-preview1-007799-linux-musl-x64.tar.gz

**Comment by weshaggard** [comment]
test ci please

**Comment by weshaggard** [comment]
Alpine test leg is expected to fail because of the renames.

https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1614252&_a=summary - linux-musl build with skiptests=true
https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1614318&_a=summary - linux-musl build with skiptests=false (not the publish failed because of the earlier successful publish).

**Comment by weshaggard** [comment]
@livarcocc This is ready to merge.

FYI @mmitche this will be a new CLI leg for orchestration, we will also need to remove the alpine build leg.

**Comment by mmitche** [comment]
Updated prodcon

**Pull request merged**